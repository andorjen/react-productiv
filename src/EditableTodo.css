.edit {
    display: inline-block;
}

.eachTodo {
    display: flex;
    justify-content: space-between;
}