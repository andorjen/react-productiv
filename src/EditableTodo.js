import React, { useState } from "react";
import Todo from "./Todo";
import TodoForm from "./TodoForm";
import "./EditableTodo.css";

/** Show editable todo item.
 *
 * Props
 * - todo
 * - update(): fn to call to update a todo
 * - remove(): fn to call to remove a todo
 * 
 * State: isEditing (true/false; initial state = false)
 *
 * EditableTodoList -> EditableTodo -> { Todo, TodoForm }
 */

function EditableTodo({ todo, update, remove }) {

  const [isEditing, setIsEditing] = useState(false);

  /** Toggle if this is being edited */
  function toggleEdit() {
    setIsEditing(isEditing=>!isEditing);
  }

  /** Call remove fn passed to this. */
  function handleDelete() {
    remove(todo.id);
  }

  /** Edit form saved; toggle isEditing and update in ancestor. */
  function handleSave(formData) {
    toggleEdit();
    update(formData);
    //Come back to this after working on the form
  }

  return (
    <div className="EditableTodo">

      {isEditing &&
        <TodoForm initialFormData={todo} handleSave={handleSave} />
      }

      {!isEditing &&
        <div className="mb-3 eachTodo">
          <Todo 
            id={todo.id}
            title={todo.title}
            description={todo.description}
            priority={todo.priority} />
          <div className="float-right text-sm-right edit">
            <button
              className="EditableTodo-toggle btn-link btn btn-sm"
              onClick={toggleEdit}>
              Edit
            </button>
            <button
              className="EditableTodo-delBtn btn-link btn btn-sm text-danger"
              onClick={handleDelete}>
              Del
            </button>
          </div>
        </div>}

    </div>
  );
}


export default EditableTodo;
