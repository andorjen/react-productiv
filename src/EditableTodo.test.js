import React from "react";
import { render, fireEvent } from "@testing-library/react";
import EditableTodo from "./EditableTodo";

const testTodo = {
    id: 1,
    title: "Code!",
    description: "Write some code",
    priority: 2
};

it("renders without crashing", function () {
    render(<EditableTodo todo={testTodo} />);
  });


it("renders the correct content before and after toggling edit button ", function () {
    const {container, debug} = render(<EditableTodo todo={testTodo} />);
    // debug(container);
    expect(container).toContainHTML("Code!");
    expect(container).toContainHTML("Write some code");
    expect(container).toContainHTML("(priority: 2)");
    expect(container).toContainHTML("Edit");
    expect(container).toContainHTML("Del");

    expect(container).not.toContainHTML('class="form-group"');
    expect(container).not.toContainHTML('class="NewTodoForm"');
    // debug(container);
    fireEvent.click(container.querySelector(".EditableTodo-toggle"));
    // debug(container);
    expect(container).toContainHTML('class="form-group"');
    expect(container).toContainHTML('class="NewTodoForm"');

});


it("matches snapshot", function () {
    const { container } = render(<EditableTodo todo={testTodo} />);
    expect(container).toMatchSnapshot();
});