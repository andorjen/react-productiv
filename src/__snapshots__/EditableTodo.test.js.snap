// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="EditableTodo"
  >
    <div
      class="mb-3 eachTodo"
    >
      <div
        class="Todo"
        id="1"
      >
        <div>
          <b>
            Code!
          </b>
           
          <small>
            (priority: 
            2
            )
          </small>
        </div>
        <div>
          <small>
            Write some code
          </small>
        </div>
      </div>
      <div
        class="float-right text-sm-right edit"
      >
        <button
          class="EditableTodo-toggle btn-link btn btn-sm"
        >
          Edit
        </button>
        <button
          class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
        >
          Del
        </button>
      </div>
    </div>
  </div>
</div>
`;
